src/snowflake/snowpark/_internal/analyzer/cte_utils.py

```diff
@@ -119,3 +131,14 @@ def build_plan_to_query_map_in_post_order(node: "SnowflakePlan") -> None:
     )
     final_query = with_stmt + SPACE + plan_to_query_map[node]
     return final_query
+
+
+def encode_id(
+    query: str, query_params: Optional[Sequence[Any]] = None
+) -> Optional[str]:
+    string = f"{query}#{query_params}" if query_params else query
+    try:
+        return hashlib.sha256(string.encode()).hexdigest()[:10]
+    except Exception as ex:
```

Comment:
I feel this protection will hide problems and cause bigger problems because queries have the same encoded id.